src/lib/Microsoft.Health.Fhir.Ingest.Validation/MappingValidator.cs

```diff
@@ -168,23 +222,27 @@ private void CheckForTemplateCompatibility(IContentTemplate contentTemplate, ILo
                         {
                             if (!availableFhirValueNames.Contains(v.ValueName))
                             {
-                                validationResult.CaptureWarning($"The value [{v.ValueName}] in Device Mapping [{extractor.Template.TypeName}] is not represented within the Fhir Template of type [{innerTemplate.TypeName}]. Available values are: [{availableFhirValueNamesDisplay}]. No value will appear inside of Observations.");
+                                validationResult.CaptureWarning(
+                                    $"The value [{v.ValueName}] in Device Mapping [{extractor.Template.TypeName}] is not represented within the Fhir Template of type [{innerTemplate.TypeName}]. Available values are: [{availableFhirValueNamesDisplay}]. No value will appear inside of Observations.",
+                                    ValidationCategory.FHIRTRANSFORMATION);
                             }
                         }
                     }
                 }
                 catch (TemplateNotFoundException)
                 {
-                    validationResult.CaptureWarning($"No matching Fhir Template exists for Device Mapping [{extractor.Template.TypeName}]. Ensure case matches. Available Fhir Templates: [{availableFhirTemplates}].");
+                    validationResult.CaptureWarning(
+                        $"No matching Fhir Template exists for Device Mapping [{extractor.Template.TypeName}]. Ensure case matches. Available Fhir Templates: [{availableFhirTemplates}].",
+                        ValidationCategory.FHIRTRANSFORMATION);
                 }
                 catch (Exception e)
                 {
-                    validationResult.CaptureException(e);
+                    validationResult.CaptureException(e, ValidationCategory.FHIRTRANSFORMATION);
                 }
             }
         }
 
-        private void ProcessDeviceEvent(JToken deviceEvent, IContentTemplate contentTemplate, DeviceResult validationResult)
+        protected virtual void ProcessDeviceEvent(JToken deviceEvent, IContentTemplate contentTemplate, DeviceResult validationResult)
```

Comment:
Did you make this change so that a derived class can leverage ProcessDeviceEvent() and ProcessNormalizedEvent()? Just curious to learn more

Reply:
Correct. I was originally going to create a subclass of the MappingValidator which would perform aggregated validations. But after thinking more I simply kept that logic inside of this class.

Reply:
Makes sense, thanks. Feel free to leave it as is